### PR TITLE
Removed dependency on topological_inventory-collector-openshift

### PR DESCRIPTION
**Pull request by slemrmartin**
Basically `Openshift::Collector` and `Openshift::Parser` moved to `MockCollector::Collector`  and `MockCollector::Parser`

### CODE REVIEW

File: lib/mock_collector/collector.rb [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+  class Collector
+    def initialize(source, _config, default_limit: 100, poll_time: 30)
+      # self.connection_manager = MockCollector::Server.new
+      self.collector_threads  = Concurrent::Map.new
+      self.finished           = Concurrent::AtomicBoolean.new(false)
+      self.limits             = Hash.new(default_limit)
+      self.log                = Logger.new(STDOUT)
+      self.openshift_host     = nil # openshift_host
+      self.openshift_port     = nil # openshift_port
+      self.openshift_token    = nil # openshift_token
```
Comment (agrare) [inline lib/mock_collector/collector.rb]: We probably don't need these three

**Comment by agrare** [comment]
@slemrmartin I think we can dramatically simplify this if we don't try to totally mimic the openshift collector, e.g. no threads and no queue just send the mock data.

File: lib/mock_collector/collector.rb (hunk 1 above)
Comment (slemrmartin) [reply lib/mock_collector/collector.rb]: @agrare yeah, I forgot to add WIP before I went home yesterday :)

**Comment by slemrmartin** [comment]
> @slemrmartin I think we can dramatically simplify this if we don't try to totally mimic the openshift collector, e.g. no threads and no queue just send the mock data.

@agrare I agree but I would like to remove dependency asap, because collector is often broken. It can be simplified in step 2

**Comment by slemrmartin** [comment]
@agrare partially simplified, I think it's ok for step 1. Tested, working

File: lib/mock_collector/collector.rb (hunk 1 above)
Comment (agrare) [reply lib/mock_collector/collector.rb]: Ah okay no worries!

**Review by agrare — APPROVED** [review]
:+1:

**Pull request merged**